probe-rs/src/architecture/arm/core/exception_handling/armv8m.rs

```diff
@@ -0,0 +1,477 @@
+use crate::{
+    core::{ExceptionInfo, ExceptionInterface},
+    debug::DebugRegisters,
+    memory::MemoryInterface,
+    memory_mapped_bitfield_register, CoreInterface, Error, MemoryMappedRegister, RegisterValue,
+};
+use bitfield::bitfield;
+
+use super::armv6m_armv7m_shared::{Xpsr, EXCEPTION_STACK_REGISTERS};
+
+bitfield! {
+    /// The EXC_RETURN value (The value of the link address register) is used to
+    /// determine the stack to return to when returning from an exception.
+    struct ExcReturn(u32);
+    /// If the value is 0xFF, then this is a valid EXC_RETURN value.
+    is_exception_flag, _: 31, 24;
+    /// Indicates whether to restore registers from the secure stack or the unsecure stack
+    use_secure_stack, _:6;
+    /// Indicates wheteher the default stacking rules apply or the callee registers are already on the stack
+    use_default_register_stacking, _:5;
+    /// Defines whether the stack frame for this exception has space allocated for FPU state information. Bit [4] is 0 if stack space is the exended frame that includes FPU registes.
+    use_standard_stackframe, _: 4;
+    /// Indicates whether the return mode is Handler (0) or Thread (1)
+    mode, _: 3;
+    /// Indicates which stack pointer the exception frame resides on
+    stack_pointer_selection, _: 2;
+    /// Indicates whether the security domain the exception was taken to was non-secure (0) or secure (1)
+    exception_secure, _: 0;
+}
+
+memory_mapped_bitfield_register! {
```

Comment:
HSFR & CSFR are identical to how they are implemented in armv6m_armv7m_shared.rs. Thus these registers and a lot of the supporting code could be reused from that module. I wasn't sure if the better approach is to have it here, or rename that module? I eventually decided to pull it into this new file, since the implementation of a lot of the supporting code needed to be modified to account for SecureFaults. But I'm not open to reorganizing some of this if that makes more sense.

Reply:
I think splitting into a new file is fine :) Generally the different impls have quite some differences and trying to unify them can be quite tricky, albeit of course very desirable :)